File: src/Domain/Entities/Challenges/Challenge.cs
using Domain.Entities.Common;
using Domain.Enums;

namespace Domain.Entities.Challenges;

public class Challenge : EntityWithId
{
    public string Name { get; set; }

    public string DescriptionShort { get; set; }

    public string DescriptionMarkdown { get; set; }

    public string StubGeneratorInput { get; set; }

    public ChallengeStatus Status { get; set; }

    public string StatusReason { get; set; }

    public List<TestPair> Tests { get; set; }

    public Solution Solution { get; set; }

    public List<string> TagIds { get; set; }

    public List<Feedback> Feedbacks { get; set; }

    public float Difficulty { get; set; }

    public string CreatedByUserId { get; set; }

    public DateTime LastModifiedOn { get; set; }
}

File: src/react-frontend/src/app/types/models/challenge/challenge.ts
import { ChallengeStatus } from 'app/types/enums/challengeStatus';
import { Solution } from '../general/solution';
import { UserDto } from '../user/userDto';
import { ChallengeTag } from './challengeTag';
import { Feedback } from './feedback';
import { TestPair } from './testPair';

export type Challenge = {
  id: string;
  name: string;
  descriptionShort: string;
  descriptionMarkdown: string;
  stubGeneratorInput: string;
  status: ChallengeStatus;
  statusReason: string;
  tests: TestPair[];
  solution: Solution;
  tags: ChallengeTag[];
  feedbacks: Feedback[];
  difficulty: number;
  fun: number;
  testCasesRelevancy: number;
  user: UserDto;
  lastModifiedOn: Date;
};
